src/Core/Framework/Validation/DataBag/DataBag.php

```diff
@@ -14,11 +14,7 @@ class DataBag extends ParameterBag
      */
     final public function __construct(array $parameters = [])
     {
-        foreach ($parameters as $key => $value) {
-            if (\is_array($value)) {
-                $parameters[$key] = new static($value);
-            }
-        }
+        $parameters = $this->wrapArrayParameters($parameters);
```

Comment:
Isn't this a break? Before that I got an array, now I get an object.

Reply:
Yes, but most bugfixes are breaking changes :)

Reply:
That clone is fine, but we can't change in a minor how the databag works with arrays. Can you do this only with 6.6 feature flag check?